wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift

```diff
@@ -21,23 +21,18 @@ import WireDataModel
 import WireSyncEngine
 
 final class DeviceDetailsViewActionsHandler: DeviceDetailsViewActions, ObservableObject {
-    let logger = WireLogger.e2ei
-    let e2eIdentityProvider: E2eIdentityProviding
-    let userSession: UserSession
-    let mlsProvider: MLSProviding
-
-    var userClient: UserClient
-    var clientRemovalObserver: ClientRemovalObserver?
-    var credentials: ZMEmailCredentials?
-    var isProcessing: ((Bool) -> Void)?
-
-    var isMLSEnabled: Bool {
-        mlsProvider.isMLSEnbaled
-    }
+    private let logger: LoggerProtocol
+    private let mlsClientResolver: MLSClientResolving
+    private let conversationId: Data?
+    private var userClient: UserClient
+    private var userSession: UserSession
+    private var clientRemovalObserver: ClientRemovalObserver?
+    private var credentials: ZMEmailCredentials?
+    private let getE2eIdentityEnabled: GetIsE2EIdentityEnabledUseCaseProtocol
+    private let getE2eIdentityCertificates: GetE2eIdentityCertificatesUseCaseProtocol
+    private let getProteusFingerprint: GetUserClientFingerprintUseCaseProtocol
 
-    var isE2eIdentityEnabled: Bool {
-        e2eIdentityProvider.isE2EIdentityEnabled
-    }
+    var isProcessing: ((Bool) -> Void)?
```

Comment:
This looks like an uncommon pattern for providing status information.
What about having a delegate or a published property?

Reply:
I think its more of an opinion. What are the pitfalls of using this pattern?